### PR TITLE
checking the version when update a resource with grpc

### PR DESCRIPTION
**Pull request by skeeey**
refer to https://github.com/openshift-online/maestro/issues/146

### CODE REVIEW

**Review by machi1990 — COMMENTED** [review]
left a small nitpick.
otherwise lgtm

The client can use the `common.CloudEventsResourceVersionAnnotationKey` annotation to make sure that the version is set. I'll give this a spin once merged

File: test/e2e/pkg/sourceclient_test.go [hunk 1]
```diff
@@ -28,6 +26,54 @@ import (
...
+			err := workClient.ManifestWorks(consumer.Name).Delete(ctx, workName, metav1.DeleteOptions{})
+			Expect(err).ShouldNot(HaveOccurred())
+
+			Eventually(func() error {
+				return AssertWorkNotFound(workName)
+			}, 30*time.Second, 1*time.Second).ShouldNot(HaveOccurred())
+
+		})
+
+		It("Should has an error when updating an obsolete work", func() {
```
[the remark this answers is not in the document]
Comment (skeeey) [reply test/e2e/pkg/sourceclient_test.go]: updated

**Comment by machi1990** [comment]
lgtm

**Comment by skeeey** [comment]
update the sdk-go to remove the version annotation dependence

**Review by clyang82 — APPROVED** [review]
LGTM

**Pull request merged**